woocommerce/woocommerce-ios
# [IAP] M2 expandable card feature details

Closes: #10199 

## Description
This PR adds the list of main features for Essential and Performance plans to the existing expandable card. For this, we start by renaming the previous `WooPlan` to `LegacyWooPlan`, since this struct currently contains a lot of unnecessary information for M2, and create a new WooPlan with the minimum necessary.

## Changes
- Renames  `WooPlan` to `LegacyWooPlan`: As with M1, the plan features data here is hardcoded since we plan to load this dynamically via the API in future milestones (~M4), and remove all legacy classes that we no longer need.
- Creates a new `WooPlanCardFeaturesView`, which displays the features of each plan

## Testing instructions

On a Free Trial site, eligible for IAP (a new site can be created in Menu > Switch Sites > Add New Site) tap on Purchase Now and observe that:
- Tapping on `View Essential features` shows the list of features in the screenshot below
- Tapping on `View Performance features` shows the text "_Everything in essential, plus:_" at the top, plus a different set of features (screenshot below).

| Essential features | Performance features |
|--------|--------|
| ![Simulator Screen Shot - iPhone 13 Pro - 2023-07-12 at 15 34 33](https://github.com/woocommerce/woocommerce-ios/assets/3812076/0ba4814e-e807-4f0a-8da3-46dc679e67cf) | ![Simulator Screen Shot - iPhone 13 Pro - 2023-07-12 at 15 34 41](https://github.com/woocommerce/woocommerce-ios/assets/3812076/fe500fed-c710-4663-a1c8-4f6dd436d02b)  |
